src/adams_methods.jl

```diff
@@ -0,0 +1,636 @@
+
+            # Set CE_loops to be how many correction-evaulation cycles
+            # desired. As least one loop is necessary, for the sake of the
+            # adaptive step selection
+            CE_loops = 1
+            loop_counter = 0
+            if n>=steporder+1
+                for loop_counter=1:CE_loops
+                    # (C)orrection of y_np1
+                    # Add new coefficients  based on this prediction
+                    # computation of ϕ, ϕstar using recurrence relations
+                    ϕ_coefs!(ϕ, ϕstar,dy_np1,1:steporder+1)
+
+                    # Calculation of g_{steporder + 1}(n) coefficients using
+                    # recurrence relations has already been done above, and does
+                    # not depend on dy_np1 or y_np1, so no update necessary
+
+                    # We correct the computation of y_np1 using y_np1
+                    # as part of our prediction Harier equation 5.7
+
+                    for i=1:dof
+                        y_np1[i] += dt*g[steporder+1]*ϕ[3,steporder+1][i]
+                    end
+
+                    # (E)valuate estimate for next derivative again
+                    dy_np1=F(t_np1,y_np1)
+                end
+            end
+            #---################################################################
+            # Error Check for step size `dt` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_dt` for next step
+            #     ->if no, use this `next_dt`` for next attempt at this step
+            #---################################################################
+
+            #Don't allow increases of steps until n >= steporder + 2
+            if n==1
+                #TODO: adaptive step selection for n<steporder +2
+                successful_step = true
+                next_dt = dt/4
+            elseif n<steporder+2
+                for i = 1:dof
+                    yerr_k[i] = dt*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1][i]
+                end
+                err_k, next_dt,timeout_VS = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_k, steporder,
+                                timeout_VS, dof, abstol, reltol, maxstep, norm)
+                if err_k <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = .5*next_dt
+                end
+            else
+                # We calculate local truncation error, LE_{k+1}; the difference
+                # between implicit results for order k and k+1.
+
+                # We need ϕ_{k+1}(n+1) and g_{k+1}(n);the latter has already
+                # been calculated above, the former still needs calculation
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+2)
+
+                # Calculate ||LE_{k+1}|| and next stepsize
+                for i = 1: dof
```

Comment:
This `yerr_kp1` gets overwritten in `stepsize_hw92!`.  If this loop is supposed to do something, then it doesn't do it, otherwise delete.  In the other places where `stepsize_hw92!` gets called, the same occurs.